### ISSUE TITLE
 Feature request: automatically trim indentation

### ISSUE THREAD

**Issue by DrKGD** [NarutoXY/silicon.lua#7]
Hi there, first and foremost thanks for the great plugin, I tend to share lots of snippets of code to my university colleagues!

I am kind of interested in an `autogobble` functionality, don't really know whether is it more appropriate to discuss it there or just ask for it on the silicon binary repository: in doubt I am going to also mention it there.
https://github.com/Aloxaf/silicon/issues/213#issue-1431899276

In case you are wondering what autogobble is (name comes from `lstautogobble`, a package from LaTeX) , look at the following images
![image](https://user-images.githubusercontent.com/44092476/199755476-d54c24cf-6ca4-4c4b-b2ad-19077382351d.png)
![image](https://user-images.githubusercontent.com/44092476/199755590-4d055f4d-d919-4a15-944d-d1039bebc8f3.png)

Basically it consists of _recalculating the indentation at the start of the line_, for each line, thus trims the excess.

**Comment by NarutoXY** [comment]
Done
